crates/sys/src/message_queue.rs

```diff
@@ -0,0 +1,30 @@
+//! Low-level bindings to Furi message queue API
+
+use core::ffi::c_void;
+
+use crate::opaque;
+
+opaque!(FuriMessageQueue);
+
+extern "C" {
+    #[link_name = "furi_message_queue_alloc"]
+    pub fn alloc(count: u32, size: usize) -> *const FuriMessageQueue;
+
+    #[link_name = "furi_message_queue_free"]
+    pub fn free(queue: *const FuriMessageQueue);
+
+    #[link_name = "furi_message_queue_put"]
+    pub fn put(queue: *const FuriMessageQueue, payload: *const c_void, timeout: u32) -> u32;
+
+    #[link_name = "furi_message_queue_get"]
+    pub fn get(queue: *const FuriMessageQueue, payload: *mut c_void, timeout: u32) -> u32;
+
+    #[link_name = "furi_message_queue_get_capacity"]
+    pub fn capacity(queue: *const FuriMessageQueue) -> u32;
```

Comment:
Normally Rust uses `usize` for count/size values.

In this case we're lucky, as `usize` on `thumbv7em-none-eabihf` is the same as a `u32` (see [`p:32:32`](https://github.com/rust-lang/rust/blob/6af09d2505f38e4f1df291df56d497fb2ad935ed/compiler/rustc_target/src/spec/thumbv7em_none_eabihf.rs#L17)).

Reply:
Happy to change it. I based this off of the header file [here](https://github.com/flipperdevices/flipperzero-firmware/blob/ec57dd310a6ad9657c08d143bd82a66da532dc78/furi/core/message_queue.h), which used `uint32_t` for the value types and `FuriStatus` as the return codes. `FuriStatus` valus are all defined as `0xFFFFFFxx`, so I inferred `u32`.